tracer/src/Datadog.Trace/Headers/IHeadersCollection.cs

```diff
@@ -17,7 +17,7 @@ internal interface IHeadersCollection
         /// </summary>
         /// <param name="name">The specified header to return values for.</param>
         /// <returns>Zero or more header strings.</returns>
-        IEnumerable<string> GetValues(string name);
+        StringValues GetValues(string name);
```

Comment:
Are we definitely handling the case where `StringValues` represents `null`? That's equivalent to returning `null` here previously I guess, which we didn't do, but which nullable ref types would flag before?

Reply:
Obsolete, but `StringValues` treats all of the following as an "empty" `StringValues`:
- null string
- empty string
- empty array
- a single-item array with a null string
- a single-item array with an empty string
- in some cases, longer arrays where all strings are null or empty (there are some exceptions, like `StringValues.ToArray()` which can return the original array)

The new `StringEnumerable` does the same.